static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -520,6 +521,15 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
       return;
     }
 
+    if (
+      name === 'dataset' &&
+      value === 'generic_metrics' &&
+      this.state.timeWindow === TimeWindow.ONE_MINUTE
+    ) {
+      // ONE MINUTE interval is not allowed for custom metrics alerts
+      this.setState({timeWindow: TimeWindow.FIVE_MINUTES});
+    }
```

Comment:
Couldn't you handle it where the `alert_type` field changes? (the condition above)
```ts
if (name === 'alertType') {
  this.setState(({dataset, timeWindow}) => ({
    alertType: value as MetricAlertType,
    dataset: this.checkOnDemandMetricsDataset(dataset, this.state.query),
    // ONE MINUTE interval is not allowed for custom metrics alerts
    timeWindow: value === 'custom_metrics' && timeWindow === TimeWindow.ONE_MINUTE
      ? TimeWindow.FIVE_MINUTES
      : timeWindow 
  }));
}
```

Reply:
If possible you should reference current state values by calling setState as a function to avoid race conditions.